metrics/sender.go

```diff
@@ -46,10 +49,28 @@ type applicationInfo struct {
 	Version string `json:"version"`
 }
 
+type natMappingFailContext struct {
+	error string
```

Comment:
please don't have a prop called error with string for type. Either call it errorInfo, or use error as type.